OrbitQt/AccessibilityAdapter.h

```diff
@@ -0,0 +1,188 @@
+                                          |                          |
+                                          |                          |
++-----------+   +----------------------+  |                          |  Adapter for child()
+| GlCanvas  +<--+GlAccessibleInterface +<----------------------------+
++-----------+   +--+-------------+-----+  |
+                   |             ^ Parent |
+                   v Child(i)    |        |
++-----------+   +--+-------------+-----+  |                    +-----------------------------+
+| TimeGraph +<--+GlAccessibleInterface +<----------------------+A11yAdapter                  |
++-----------+   +----------------------+  |                    +-----------------------------+
+                                          |
+                                          +
+
+The OrbitGlWidgetAccessible is automatically created when an OrbitGlWidget is constructed by
+installing a QT Accessibility Factory (InstallAccessibilityFactories()).
+
+To make sure adapters created by A11yAdapter::GetOrCreateAdapter are deleted when the corresponding
+GlAccessibleInterface is deleted, these interfaces register themselves in the
+GlAccessibleInterfaceRegistry (see OrbitGlAccessibility.h), which in turn allows to register a
+callback on interface deletion. A11yAdapter registers itself for this callback and cleans up
+interfaces accordingly.
+**/
+
+#ifndef ORBIT_QT_GL_ACCESSIBILITY_H_
+#define ORBIT_QT_GL_ACCESSIBILITY_H_
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/container/flat_hash_set.h>
+
+#include <QAccessible>
+#include <QAccessibleWidget>
+#include <QWidget>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitGlAccessibility.h"
+#include "orbitglwidget.h"
+
+namespace orbit_qt {
+
+/*Instances of this act as adapters from QAccessibleInterface to orbit_gl::GlAccessibleInterface and
+ vice versa. Static methods of A11yAdapter provide factory methods and tracking of adapters.
+
+ See file documentation above for more details.
+ */
+class A11yAdapter : public QAccessibleInterface {
+ public:
+  A11yAdapter() = delete;
+  A11yAdapter(A11yAdapter& rhs) = delete;
+  A11yAdapter(A11yAdapter&& rhs) = delete;
+
+  // vvv Overrides QAccessibleInterface vvv
+  bool isValid() const override {
+    bool result = info_ != nullptr;
+    CHECK(!result || interface_map_.find(info_)->second == this);
+    return result;
+  }
+  QObject* object() const override { return &dummy_; }
+  QAccessibleInterface* focusChild() const override { return nullptr; }
+
+  QAccessibleInterface* parent() const override {
+    return GetOrCreateAdapter(info_->AccessibleParent());
+  }
+  QAccessibleInterface* child(int index) const override {
+    return GetOrCreateAdapter(info_->AccessibleChild(index));
+  }
+  int childCount() const override { return info_->AccessibleChildCount(); }
+  int indexOfChild(const QAccessibleInterface* child) const override;
+  [[nodiscard]] QAccessibleInterface* childAt(int x, int y) const override;
```

Comment:
Why is this `[[nodiscard]]` and the others aren't?

Reply:
Good question. I removed the `[[nodiscard]]` since this method is defined by the `QAccessibleInterface` without it.